### ISSUE TITLE
Ignore structs with doc(hidden) in exhaustive_structs lint

### ISSUE THREAD

**Issue by DevinR528** [rust-lang/rust-clippy#7349]
Lint name: `exhaustive_structs`

I tried this code:

```rust
#[doc(hidden)]
pub struct Foo;
```

I expected to see this happen: No warning when the `exhaustive_structs` lint is enabled. The lint could/should consider structs with `doc(hidden)` private?

Instead, this happened: The lint emitted a warning.

### Meta
  ```
clippy 0.1.54 (35fff69 2021-06-06)

rustc 1.54.0-nightly (35fff69d0 2021-06-06)
binary: rustc
commit-hash: 35fff69d043b1c0f5c29894e7f4b0da8b039c131
commit-date: 2021-06-06
host: x86_64-unknown-linux-gnu
release: 1.54.0-nightly
LLVM version: 12.0.1
  ```
